File: JavaScript/kindergarten-garden.js
const DEFAULT_STUDENTS = [
    'Alice',
    'Bob',
    'Charlie',
    'David',
    'Eve',
    'Fred',
    'Ginny',
    'Harriet',
    'Ileana',
    'Joseph',
    'Kincaid',
    'Larry',
];

const PLANT_CODES = {
    G: 'grass',
    V: 'violets',
    R: 'radishes',
    C: 'clover',
};

export class Garden {
    constructor(diagram, students = DEFAULT_STUDENTS) {
        this.rows = diagram.split("\n");
        this.students = students;
    }

    plants(student) {
        let index = this.students.sort().indexOf(student) * 2;

        let plants = [
            this.rows[0][index], this.rows[0][index + 1],
            this.rows[1][index], this.rows[1][index + 1],
        ].map(p => PLANT_CODES[p]);

        return plants;
    }
}

let garden = new Garden("VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV");

console.log(garden.rows);

DEFAULT_STUDENTS.forEach(student => {
    console.log(student + "->" + garden.plants(student));
});

File: C#/Kindergarten Garden/KindergartenGarden.cs
using System;
using System.Collections.Generic;

public enum Plant
{
    Violets,
    Radishes,
    Clover,
    Grass
}

public static class PlantExtensions
{
    public static Plant ToPlant(this char plantCode)
    {
        return plantCode switch
        {
            'V' => Plant.Violets,
            'R' => Plant.Radishes,
            'C' => Plant.Clover,
            'G' => Plant.Grass,
            _ => throw new ArgumentException($"Invalid plant code: {plantCode}")
        };
    }
}

public class KindergartenGarden
{
    public static readonly List<string> DEFAULT_STUDENTS = new List<string>() {
        "Alice",
        "Bob",
        "Charlie",
        "David",
        "Eve",
        "Fred",
        "Ginny",
        "Harriet",
        "Ileana",
        "Joseph",
        "Kincaid",
        "Larry",
    };

    public string[] rows { get; }

    public KindergartenGarden(string diagram)
    {
        this.rows = diagram.Split("\n");
    }

    public IEnumerable<Plant> Plants(string student)
    {
        int index = DEFAULT_STUDENTS.IndexOf(student) * 2;

        var plants = new List<Plant>
        {
            this.rows[0][index].ToPlant(),
            this.rows[0][index+1].ToPlant(),
            this.rows[1][index].ToPlant(),
            this.rows[1][index+1].ToPlant(),
        };
        
        return plants;
    }
}